source/Lib/DecoderLib/BinDecoder.cpp

```diff
@@ -306,60 +313,26 @@ unsigned BinDecoder::decodeBin( unsigned ctxId )
   Range       |= ( b & LPS );
   Range      <<= numBits;
   
-  // b    0 0 1 1
-  // bin  0 1 0 1
-  // res  0 1 1 0
-  
-  //bin          = ( ~b & bin ) | ( b & !bin );
   bin         ^= b;
   bin         &= 1;
   
   bitsNeeded  += numBits & ( a | b );
   
-  if( bitsNeeded >= 0 )
-  {
-    Value      += m_Bitstream->readByte() << bitsNeeded; 
-    bitsNeeded -= 8;
-  }
-
-  //if( Value < SR )
-  //{
-  //  // MPS path
-  //  if( Range < 256 )
-  //  {
-  //    int numBits   = rcProbModel.getRenormBitsRange( Range );
-  //    Range       <<= numBits;
-  //    Value       <<= numBits;
-  //    bitsNeeded   += numBits;
-  //    if( bitsNeeded >= 0 )
-  //    {
-  //      Value      += m_Bitstream->readByte() << bitsNeeded; 
-  //      bitsNeeded -= 8;
-  //    }
-  //  }
-  //}
-  //else
-  //{
-  //  bin = !bin;
-  //  // LPS path
-  //  int numBits   = rcProbModel.getRenormBitsLPS( LPS );
-  //  Value        -= SR;
-  //  Value       <<= numBits;
-  //  Range         = LPS     << numBits;
-  //  bitsNeeded   += numBits;
-  //  if( bitsNeeded >= 0 )
-  //  {
-  //    Value      += m_Bitstream->readByte() << bitsNeeded;
-  //    bitsNeeded -= 8;
-  //  }
-  //}
```

Comment:
Please don't delete comments. I like to have the unoptimized code as a documentation next to highly optimized parts, especially when doing stuff like branchless, which might be hard to read,

Reply:
The unoptimized code has been added back.